clang/test/CodeGenSYCL/kernelname-enum.cpp

```diff
@@ -0,0 +1,157 @@
+// RUN: %clang_cc1 -I %S/Inputs -fsycl -fsycl-is-device -triple spir64-unknown-unknown-sycldevice -fsycl-int-header=%t.h %s -fsyntax-only
+// RUN: FileCheck -input-file=%t.h %s
+
+#include "sycl.hpp"
+
+enum unscoped_enum : int {
+  val_1,
+  val_2
+};
+
+enum unscoped_enum_no_type_set {
```

Comment:
This type is not forward declarable, the produced integration header will cause a compilation error in the host phase https://godbolt.org/z/qH7Kjr

Reply:
@elizabethandrews, this must be addressed before the merge.
Since the PR is going to be updated, it would be nice if you address [this comment ](https://github.com/intel/llvm/pull/1675#discussion_r425635524) as well.

Reply:
@Bader: She and I have been discussing this offline, and believe that we should just prohibit use of unscoped, non-underlying type enums.  She's still working on a patch to do so.

Reply:
Yes. Will do. I'm working on it. Erich pointed out in an offline discussion that Enum Type used as template argument (Not value)  could cause issues as well and it does. I'm working on fixing it.

Reply:
Got it.
The patch was approved by code owners and passed validation, so I wanted leave a note clarifying the reason why it's not merged already.

Reply:
Yeah, I should have mentioned to hold up, but I was hoping @elizabethandrews  would have something done by the time you got back to work :)